Documentation/Getting-Started-with-Modules-Part-1.markdown

```diff
@@ -0,0 +1,427 @@
+
+It will be a simple banner which you can display in your site by adding a widget via the admin dashboard. This will be enough to show the core concepts of a module. We will come back and make improvements in the next two parts of this course.
+
+## Content part
+The content part class is the core data structure. When you scaffolded the module it automatically made you a `Models` folder. Now we need to add the `FeaturedProductPart` class to this folder:
+
+![](../Attachments/getting-started-with-modules-part-1/add-contentpart-class.png)
+
+  1. `Right click` on the `Models` folder.
+  
+  1. Choose `Add`
+  
+  1. Choose `Class...`
+
+  1. In the `Name:` field type `FeaturedProductPart`
+  
+  1. Click `Add`
+  
+Your new class will be created and opened up in the Visual Studio editor.
+
+> **Important note:** In order for Orchard to recognize Content Part classes they must be in a namespace ending `.Models`. 
+
+> Because you already added this class within the `Models` folder the namespace is automatically wrapped around your class. In the future, when you're making your own classes don't forget to ensure that you follow this namespace structure.
+
+Your content part class needs to derive from the `ContentPart` class. Normally you would also add all the data you want to store as public properties but as we are keeping it simple this first example won't have any.
+
+![](../Attachments/getting-started-with-modules-part-1/contentpart-add-inheritance.png)
+
+Add the `ContentPart` inheritance by following these steps:
+
+  1. Type ` : ContentPart` after your `FeaturedProductPart` class definition to inherit from the `ContentPart` class.
+  
+  1. Wait a second and the red squiggles will appear underneath the class. Add the namespace by pressing `Ctrl-.` on your keyboard to bring up the Quick Actions menu.
+  
+  1. Select the `using Orchard.ContentManagement;` option and press `enter`.
+    
+ That's all you need to do for your first ContentPart class.
+
+## Data migrations
+When your module is enabled in the admin dashboard the the data migration class runs. The purpose of the data migration class is to register the module with Orchard and list the features and data it uses.
+
+We aren't going to use this yet, but the migration is also used for upgrades. As you work on your modules you will want to add and remove bits. The data migration class can make changes and you can transform your existing data to meet your new requirements.
+
+The data migration class can be created by hand, following a similar process as the last section or we can scaffold it with the `orchard.exe` command line. Let's dive back in to the command line and add a data migration class to the module.
```

Comment:
Should this have a comma after 'the last section' ??

The data migration class can be created by hand, following a similar process as the last section **,** or we can scaffold it with the `orchard.exe` command line. Let's dive back in to the command line and add a data migration class to the module.